src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private final double high; //Add random parameters of Figure
+    private final double firstSide;
+    private final double secondSide;
+
+    private final ColorSupplier color;
```

Comment:
Don't use abstract classes to set behavior for classes. Here, ColorSupplier should not be a field of the IsoscelesTrapezoid class. Consider passing the color as a parameter to the constructor and assigning it to a Color type field instead.